scripts/release/uniprot_update/uniprot_xml2sql_isoform.pl

```diff
@@ -440,40 +441,44 @@
                 next;
             }
             $sdi->inflate();
-            
+
             my $sdd = $sdi->db_id;
             print "Updating master sequence...$sdd\t$ac\n";
-                        
+
             $sdi->Created( @{ $sdi->Created } );
             $sdi->Modified( @{ $sdi->Modified } );
             $sdi->add_attribute_value( 'modified', $instance_edit );
 
             updateinstance($sdi, \%values, $_);
-            
+
             $dupl_flag = 1;
-            
+
             ## master sequence update finished
-            
-            $values{'species'} ||= $sdi->species->[0]; #Inherit species instance (if not already specified) from isoform parent
-            foreach my $is_ac ( sort keys %isoids ) {    #isoforms update                
+
+
+            #Inherit species instance (if not already specified) from isoform parent
+            if (!(grep { defined } @{$values{'species'}})) {

```

Comment:
The condition here is rather cryptic... but maybe that's just the nature of Perl? I think this code is trying to check and see if a map has a "species" key, but I'm not sure I'm reading that correctly. Perhaps this needs to be cleaned and made clearer.

Reply:
I agree its cryptic but that its also the nature of Perl. Maybe when we get some Perlistic spots like this we should have a comment that more explicitly explains the logic?

Reply:
Sure, I'll add a comment and 3 riddles so it still has respectable crypticness :).

Reply:
What gets wetter the more that it dries?

Reply:
Haha, I think I've heard that one -- is it a towel?

Reply:
Yep its my go-to riddle